src/read/coff/import.rs

```diff
@@ -0,0 +1,208 @@
+//! Support for reading short import files.
+//!
+//! These are used by some Windows linkers as a more compact way to describe
+//! dynamically imported symbols.
+
+use crate::read::{Architecture, Error, ReadError, ReadRef, Result};
+use crate::{pe, ByteString, Bytes, LittleEndian as LE};
+
+/// A Windows short form description of a symbol to import.
+/// Used in Windows import libraries. This is not an object file.
+#[derive(Debug, Clone)]
+pub struct MsImportFile<'data> {
+    header: &'data pe::ImportObjectHeader,
+    dll: ByteString<'data>,
+    symbol: ByteString<'data>,
+    kind: ImportType,
+    import: Option<ByteString<'data>>,
+}
+impl<'data> MsImportFile<'data> {
+    /// Parse it.
+    pub fn parse<R: ReadRef<'data>>(data: R) -> Result<Self> {
+        let mut offset = 0;
+        let header = pe::ImportObjectHeader::parse(data, &mut offset)?;
+
+        let mut strings = Bytes(
+            data.read_bytes(&mut offset, header.data_size() as u64)
+                .read_error("Not enough data")?,
+        );
+        let symbol = strings
+            .read_string()
+            .read_error("Could not read symbol name")?;
+        let dll = strings
+            .read_string()
+            .read_error("Could not read DLL name")?;
+
+        // Unmangles a name by removing a `?`, `@` or `_` prefix.
+        fn strip_prefix(s: &[u8]) -> &[u8] {
+            match s.split_first() {
+                Some((b'?' | b'@' | b'_', rest)) => rest,
+                _ => s,
+            }
+        }
+        Ok(Self {
+            header,
+            dll: ByteString(dll),
+            symbol: ByteString(symbol),
+            kind: header.import_type()?,
+            import: match header.name_type()? {
+                ImportNameType::Ordinal => None,
+                ImportNameType::Name => Some(symbol),
+                ImportNameType::NameNoPrefix => Some(strip_prefix(symbol)),
+                ImportNameType::NameUndecorate => {
+                    Some(strip_prefix(symbol).split(|&b| b == b'@').next().unwrap())
```

Comment:
For both of these, have you tested that this stripping is needed? From reading the documentation, my impression was that the name stored in the file has already had this stripped.

Reply:
I used this PR to search `.lib` files on my machine for `NameUndecorate` types. I then checked the lib using the output of msvc's `dumpbin.exe`. I'll reproduce it below (truncated output to only the relevant part).

```
> dumpbin /EXPORTS /HEADERS 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x86\6.0.15\runtimes\win-x86\native\ijwhost.lib'

  Version      : 0
  Machine      : 14C (x86)
  TimeDateStamp: D1C56A4E
  SizeOfData   : 0000001C
  DLL name     : ijwhost.dll
  Symbol name  : __CorDllMain@12
  Type         : code
  Name type    : undecorate
  Hint         : 1
  Name         : _CorDllMain

     Exports

       ordinal    name

                  __CorDllMain@12
```

Compare to using the `objdump` in this PR:

```
ijwhost.dll:
Format: Short Import File
Architecture: I386
DLL: "ijwhost.dll"
Symbol: "__CorDllMain@12"
Import: Name("_CorDllMain")
Type: Code
```

Reply:
Thanks, that looks good then. I'm still a bit fuzzy on how this is all used though. Do you know why the file format is like this? I guess the consumer needs to use both the import name and the symbol name for some reason?

Reply:
Hm, think of it like the difference between a variable name and a string in a `HashMap<String, Address>`. At load time the string in the import table is used to find the address in the DLL's export table and copy it back to the import table. However, the application's code just directly references where the imported address will be copied to. The name for this reference is under its control so it does not have to match the string used in the import/export tables. Although in practice they will usually be the same.